src/wallet/use-cases/register-wallet/register-wallet.spec.ts

```diff
@@ -0,0 +1,100 @@
+import { Test, TestingModule } from '@nestjs/testing';
+import { BadRequestException, NotFoundException } from '@nestjs/common';
+
+import { RegisterWalletUseCase } from '@/wallet/use-cases';
+import { IWalletRepository } from '@/wallet/interfaces';
+import { SaveWalletDTO } from '@/wallet/dto';
+import { AccountType } from '@/shared/constants/enums';
+import { ListUserByIdUseCase } from '@/user/use-cases';
+import { FindBankByIdUseCase } from '@/bank/use-cases';
+import { Wallet } from '@/wallet/infra/entities';
+
+describe('Register Wallet Use Case', () => {
+  let registerWalletUseCase: RegisterWalletUseCase;
+  let walletRepository: IWalletRepository;
+  let listUserByIdUseCase: ListUserByIdUseCase;
+  let findBankByIdUseCase: FindBankByIdUseCase;
+  let createdUser: SaveWalletDTO;
+
+  beforeEach(async () => {
+    const module: TestingModule = await Test.createTestingModule({
+      providers: [
+        RegisterWalletUseCase,
+        {
+          provide: IWalletRepository,
+          useValue: {
+            create: jest.fn(),
```

Comment:
Uma coisa que é interessante é vocÊ manter os mocks do fluxo de sucesso aqui, ou seja, os dados que esse repository ou um service ou o que for que esteja mockando, tem que retornar no caso de sucesso.
Aqui você só está colocando o jest.fn(), é só completar com o mock de sucesso:
create: jest.fn().mockResolvedValue(dados do caso de sucesso)

dessa forma todos os testes já vão estar com esse retorno padrão e você só muda nos testes o que você quer que tenha um retorno diferente, de preferência usando o jest.spyOn().mockResolvedValueOnce() ou est.spyOn().mockImplementationOnce()

Reply:
Dei uma alterada la Luiz! Ve se bateu certinho com o que voce disse, por favor.